File: Task 4/Java/app/src/main/java/dev/ogabek/java/adapter/PostAdapter.java
package dev.ogabek.java.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import dev.ogabek.java.R;
import dev.ogabek.java.model.Post;

public class PostAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private final Context context;
    private final ArrayList<Post> posts;

    public PostAdapter(Context context, ArrayList<Post> posts) {
        this.context = context;
        this.posts = posts;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_post_list, parent, false);
        return new ItemViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        Post post = posts.get(position);
        if (holder instanceof ItemViewHolder) {
            ((ItemViewHolder) holder).tv_title.setText(post.getTitle().toUpperCase());
            ((ItemViewHolder) holder).tv_body.setText(post.getBody());
        }
    }

    @Override
    public int getItemCount() {
        return posts.size();
    }

    private static class ItemViewHolder extends RecyclerView.ViewHolder {

        TextView tv_title, tv_body;

        public ItemViewHolder(View view) {
            super(view);

            tv_title = view.findViewById(R.id.tv_title);
            tv_body = view.findViewById(R.id.tv_body);

        }
    }
}

File: Task 1/Kotlin/app/src/main/java/dev/ogabek/kotlin/adapter/PostAdapter.kt
package dev.ogabek.kotlin.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import dev.ogabek.kotlin.activity.MainActivity
import dev.ogabek.kotlin.R
import dev.ogabek.kotlin.model.Post

class PostAdapter(var activity: MainActivity, var items: ArrayList<Post>) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    override fun getItemCount(): Int {
        return items.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_post_list, parent, false)
        return PostViewHolder(view)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        val post: Post = items[position]
        if (holder is PostViewHolder) {
            Glide.with(activity).load(post.img).placeholder(R.drawable.loading01).into(holder.image)
            holder.tv_title.text = post.title!!.toUpperCase()
            holder.tv_body.text = post.body
        }
    }

    inner class PostViewHolder(var view: View) : RecyclerView.ViewHolder(view) {
        var tv_title: TextView = view.findViewById(R.id.tv_title)
        var tv_body: TextView = view.findViewById(R.id.tv_body)
        val image : ImageView = view.findViewById(R.id.iv_main)
    }

    init {
        this.activity = activity
        this.items = items
    }
}